packages/components/tests/integration/components/hds/pagination/index-test.js

```diff
@@ -1,17 +1,201 @@
 import { module, test } from 'qunit';
 import { setupRenderingTest } from 'ember-qunit';
-import { render } from '@ember/test-helpers';
+import { click, render, select } from '@ember/test-helpers';
 import { hbs } from 'ember-cli-htmlbars';
 
 module('Integration | Component | hds/pagination/index', function (hooks) {
   setupRenderingTest(hooks);
 
+  // Test rendering:
+
   test('it renders the component', async function (assert) {
-    await render(hbs`<Hds::Pagination />`);
+    await render(hbs`
+      <Hds::Pagination @totalItems={{100}} @itemsPerPage={{10}} @currentPage={{1}} as |P|>
+        <P.Nav />
+      </Hds::Pagination>
+    `);
     assert.dom(this.element).exists();
   });
+
   test('it should render with a CSS class that matches the component name', async function (assert) {
-    await render(hbs`<Hds::Pagination id="test-pagination" />`);
-    assert.dom('#test-pagination').hasClass('hds-pagination');
+    await render(hbs`
+      <Hds::Pagination @totalItems={{100}} @itemsPerPage={{10}} @currentPage={{1}} id="test-pagination-bar" as |P|>
+        <P.Nav />
+      </Hds::Pagination>
+    `);
+    assert.dom('#test-pagination-bar').hasClass('hds-pagination');
+  });
+
+  test('it renders the passed in child components', async function (assert) {
+    await render(hbs`
+      <Hds::Pagination @totalItems={{100}} @itemsPerPage={{10}} @currentPage={{1}} as |P|>
+        <P.Info id="test-pagination-info" />
+        <P.Nav id="test-pagination-nav" />
+        <P.SizeSelector @sizes={{array 10 30 50}} id="test-pagination-size-selector" />
+      </Hds::Pagination>
+    `);
+    assert.dom('#test-pagination-info').hasClass('hds-pagination-info');
+    assert.dom('#test-pagination-nav').hasClass('hds-pagination-nav');
+    assert
+      .dom('#test-pagination-size-selector')
+      .hasClass('hds-pagination-size-selector');
+  });
+
+  // Test user API:
+
+  test('the selected page in numbered pagination should match the passed in currentPage value', async function (assert) {
+    await render(hbs`
+    <Hds::Pagination @totalItems={{100}} @itemsPerPage={{10}} @currentPage={{2}} as |P|>
+      <P.Nav @type="numbered" />
+    </Hds::Pagination>
+    `);
+    assert.dom('.hds-pagination-nav__button-number--is-selected').exists();
+    assert
+      .dom(
+        '.hds-pagination-nav__button-number--is-selected .hds-pagination-nav__control'
+      )
+      .includesText('2');
```

Comment:
what if the page selected is `22` instead of `2`? probably I would go more specific in the test:
```suggestion
      .hasText('page 2');
```
(probably I would do the same for the tests below too)